### ISSUE TITLE
CVE-2024-29857 - org.bouncycastle:bc-fips

### ISSUE THREAD

**Issue by vsun757** [keycloak/keycloak#30341]
### Description

CVE-2024-29857 - org.bouncycastle:bc-fips affects Keycloak version 25.0.0

Relates to: https://github.com/keycloak/keycloak/issues/30340

**Comment by codespearhead** [comment]
[CVE-2024-29857](https://github.com/advisories/GHSA-8xfc-gm6g-vgpv) should have been fixed by https://github.com/keycloak/keycloak/pull/30174 , since [Quarkus 3.8.5 already declares org.bouncycastle:bc-fips:1.0.2.5, which is unaffected](https://github.com/quarkusio/quarkus/blob/00bf910d6617002b61822aea2652124ae2e11efa/bom/application/pom.xml#L19).

However, instead of inheriting the version from Quarkus, the version is currently hardcoded in the POM:

https://github.com/keycloak/keycloak/blob/2f11605a5b6a283e2949a97a157280ebb85a0aaf/pom.xml#L85-L86

https://github.com/keycloak/keycloak/blob/2f11605a5b6a283e2949a97a157280ebb85a0aaf/pom.xml#L390-L394

Because as @vmuzikar [pointed out](https://github.com/keycloak/keycloak/issues/26568#issuecomment-1916969703) in https://github.com/keycloak/keycloak/issues/26568 that bc-fips 1.0.2.4 causes "potential prod issues", as noted in a comment inside the Keycloak POM file.

@trixpan seems to be working on it in https://github.com/keycloak/keycloak/pull/29670 .

**Comment by vmuzikar** [comment]
@vsun757 @codespearhead We're not including BC FIPS dependencies in the Keycloak dist, but we're instructing users to install them themself. AFAIK we override the version, so it's not fixed by the Quarkus upgrade.

**Comment by abstractj** [comment]
@keycloak/cloud-native can we close this issue?

**Comment by rmartinc** [comment]
I started to check this the previous week and I realized that this is already being managed in https://github.com/keycloak/keycloak/issues/26568. We don't provide bc-fips in the bundle so this is more about testing/changing the documentation to link to bc-fips 1.0.2.5. There several changes in bc-fips that are affecting keycloak so the upgrade is not clean and it needs some changes that we need to discuss.

I'm closing this issue, the upgrade of bc-fips in testing and documentation will be managed in #26568.

Duplicate of #26568.
